### PR TITLE
125 music functionality

### PR DESCRIPTION
**Pull request by LemonFace0309**
image
image

- still need to work on q functionality and improving the playlists (called our picks right now) tab

Cheers 🍻

### CODE REVIEW

File: utils/getCookie.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const getCookie = (cookies, name) => {
+  let value = null;
+
+  const ca = cookies.split('; ');
+  ca.forEach((cookie) => {
+    const [cName, cValue] = cookie.split('=');
+    if (cName === name) value = cValue;
+  });
+  return value;
```
Comment (jimmyygit) [inline utils/getCookie.js:9]: can probably short circuit here when `cName === name` is found

File: components/Spaces/CallTabs/Music/Track.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+/* eslint-disable jsx-a11y/interactive-supports-focus */
+/* eslint-disable @next/next/no-img-element */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
```
Comment (jimmyygit) [inline components/Spaces/CallTabs/Music/Track.js:3]: why we no want this?

File: components/Spaces/CallTabs/Music/Tabs/SearchSongs.js [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+    let cancel = false;
+    spotifyApi.searchTracks(search).then((res) => {
+      if (cancel) return;
+      setSearchResults(
+        res.body.tracks.items.map((track) => {
+          const smallestAlbumImage = track.album.images.reduce((smallest, image) => {
+            if (image.height < smallest.height) return image;
+            return smallest;
+          }, track.album.images[0]);
```
Comment (jimmyygit) [inline components/Spaces/CallTabs/Music/Tabs/SearchSongs.js:47]: i think lodash has min function for this idk
https://lodash.com/docs/#minBy

**Review by jimmyygit — APPROVED** [review]
:3

File: components/Spaces/CallTabs/Music/Track.js (hunk 2 above)
Comment (LemonFace0309) [reply components/Spaces/CallTabs/Music/Track.js:3]: will add later. Too much work to do in this pr ahaha

**Pull request merged**